id: PYSEC-2020-121
package:
  name: tensorflow
  ecosystem: PyPI
summary: Vulnerability in tensorflow
details: In Tensorflow before version 2.3.1, the `SparseCountSparseOutput` implementation
  does not validate that the input arguments form a valid sparse tensor. In particular,
  there is no validation that the `indices` tensor has the same shape as the `values`
  one. The values in these tensors are always accessed in parallel. Thus, a shape
  mismatch can result in accesses outside the bounds of heap allocated buffers. The
  issue is patched in commit 3cbb917b4714766030b28eba9fb41bb97ce9ee02 and is released
  in TensorFlow version 2.3.1.
affects:
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    fixed: 3cbb917b4714766030b28eba9fb41bb97ce9ee02
  - type: ECOSYSTEM
    fixed: 2.3.1
references:
- type: WEB
  url: https://github.com/tensorflow/tensorflow/releases/tag/v2.3.1
- type: WEB
  url: https://github.com/tensorflow/tensorflow/commit/3cbb917b4714766030b28eba9fb41bb97ce9ee02
- type: WEB
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-jc87-6vpp-7ff3
aliases:
- CVE-2020-15198
modified: "2020-10-01T16:06:00Z"
published: "2020-09-25T19:15:00Z"
